testlib/infra.py

```diff
@@ -51,6 +51,10 @@ def check_result(result, format_str, format_str_args):
             return
         error_strings.append(f"{format_str % format_str_args}: {msg}")
 
+    if process_exists("stratisd") is None:
+        terminate_traces(MONITOR_DBUS_SIGNALS)
```

Comment:
I realized we can simplify further.

Delete the terminate_traces() calls everywhere they are now and put one unconditionally right at the start of the clean_up method. We're only running it for cleanup, and stopping it only if it wasn't stopped in the teardown method. No need for it to record all our cleanup actions.

Then move the check for process_exists right after the terminate_traces() call, so it is totally separate from any of the code chunks that do D-Bus based cleanups.

And then this is done.

Reply:
Should these be before or after the `exec_command(['udevadm', 'settle'])` call?

Reply:
...I chose after; it seems sensible.  Updated.